### ISSUE TITLE
BUSI dataset is much worse than the indicators reported in the original paper

### ISSUE THREAD

**Issue by t110368027** [FengheTan9/CMU-Net#1]
Thanks for this repo and paper. 
I'm attempting to reproduce the results of your paper on the BUSI dataset.
I used your code without making any changes

Ryan

## Result
- withoud seed (5 times)
F1 : 0.796034+-0.00266
IOU: 0.673886+-0.00235
- seed 42
F1: 0.797733
IOU: 0.671944
- seed 123
F1: 0.786534
IOU: 0.661375
- seed 3042
F1: 0.788083
IOU: 0.664477

## Config and Command
Train and Valid Command
```bash
python main.py --dataset BUSI --name CMUnet --img_ext .png --mask_ext .png --lr 0.0001 --epochs 300 --input_w 256 --input_h 256 --b 8
```
```
config setting
--------------------
name: CMUnet
epochs: 300
batch_size: 8
deep_supervision: False
input_channels: 3
num_classes: 1
input_w: 256
input_h: 256
loss: BCEDiceLoss
dataset: BUSI
img_ext: .png
mask_ext: .png
optimizer: Adam
lr: 0.0001
momentum: 0.9
weight_decay: 0.0
nesterov: False
scheduler: CosineAnnealingLR
min_lr: 1e-05
factor: 0.1
patience: 2
milestones: 1,2
gamma: 0.6666666666666666
early_stopping: -1
num_workers: 4
--------------------
```

## Environment
os: Ubuntu 20
GPU: V100 32G
pytorch 1.11.0
cudatoolkit: 11.3.1

**Comment by FengheTan9** [comment]
Thanks for using our code. 

We think that is caused by random seed in the function of train_test_split. Our code based on the UNeXt (https://github.com/jeya-maria-jose/UNeXt-pytorch) and we use the same seed to train UNext to achieve its accuracy as described in the article. 

You can first try to train UNext to the same performance as the article and use the same environment to train our code.

Our Environment

- GPU: Tesla T4
- Pytorch: 1.11.0   cuda 11.3
- cudatoolkit: 11.3.1
- scikit-learn: 1.0.2

or

- GPU: NVIDIA GeForce RTX4090 GPU
- Pytorch: 1.13.0 cuda 11.7
- cudatoolkit: 11.7.1
- scikit-learn: 1.0.2

**Comment by t110368027** [comment]
It seems like it's a seed issue in sklearn
`train_img_ids, val_img_ids = train_test_split(img_ids, test_size=0.2, random_state=41)`
I will try other seeds and observe the validation results.

Another question is that I see that the benchmark of the BUSI dataset in the UNeXt paper didn't have the same results in other models, e.g. TransUNet, U-Net, U-Net++.
These three models have achieved better results in your paper.
Is there any differences in hyper parameters?

**Comment by FengheTan9** [comment]
We use the same hyperparameter on all methods.
The reference code of U-Net and TransUnet are followed:
- [U-Net](https://github.com/LeeJunHyun/Image_Segmentation) 
- [TransUnet](https://github.com/Beckschen/TransUNet)

**Comment by Studentpengyu** [comment]
How do you handle multiple masks corresponding to one image, as shown in the picture? I simply merged them. Is that correct?
![7517b84076b31eef7b78b7f5bd6146e](https://github.com/user-attachments/assets/f0fe2f0c-60fa-4144-adf0-7a4aa6a8e6db)
